### DISCUSSION TITLE
How to generate a png/jpg image out of a design

### DISCUSSION THREAD

**Discussion by ansamassidd** [ansys/pyansys-geometry#1540] [Q&A] [answered]
Hello Community,

I am using a `Modeler` object (generated by `launch_modeler_with_spaceclaim`) to create a design (with extruded sketches) and export as a _stp_ file. So far so good.

I wonder if there is a programmatic way to generate an enjoyable png/jpg image out of a design. 
Actually I am able to generate a _png_ by simply calling the `save` method on a `Design` object. The problem is that the generated image, in many cases, doesn't render the solid effectively (possibly because the point of view is shifted or the zoom is excessive).

Any insight?

**Comment by RobPasMue** [comment, marked as answer]
Hi @ansamassidd! The ``Design``, ``Component`` and ``Body`` objects have the option to take a screenshot of your 3d body using PyVista's capabilities. For example:

```py
design.plot(screenshot="result.png")
```

This will render a 3D view of your design and store it in a screenshot. The view will be always in Isometric mode, and that's probably something we can enhance...

Another option is to use the screenshot button directly in the plotter. If you run the following:

```py
design.plot()
```

And then change your view/zoom as desired with your mouse, you can finally click on the "Screenshot" button to save it as a PNG file as well. In the image, I framed the screenshot button in a red box for visibility.

![image](https://github.com/user-attachments/assets/bae2c8b0-8336-47e7-8f7b-134510f9bc85)


Hope all this helps!

**Reply by RobPasMue** [reply]
By the way, if this reply answered your doubts, please mark it as an answer so that other users can benefit from the response =)

**Comment by ansamassidd** [comment]
Hi @RobPasMue, 

thanks a lot for your reactive and exhaustive response. 
Your first piece of advice well suites my use case, because I need the screenshot to be generated programmatically.

The file that gets saved on my filesystem is exactly what I was looking for. 

However, I see that a `pyvista` window pops up for a short amount of time (<1s). Is it possible to avoid this side effect?

**Reply by RobPasMue** [reply]
Yes, that is possible. Add the following lines at the top of your script:

```py
import pyvista as pv

pv.OFF_SCREEN = True
```

This will prevent PyVista from popping up =)

**Reply by ansamassidd** [reply]
It works perfectly. Thank you Roberto!

**Reply by RobPasMue** [reply]
Awesome, glad it helped!
